src/workerd/server/tests/compile-tests/compile-test.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# $1 -> workerd binary path
+# $2 -> path to desired file to compile
+# $3 -> port to curl
+# $4 -> desired output
+
+# Help Function
+function show_help() {
+  echo "
+usage: compile-test.sh [-d] [-h] <workerd command> <file-to-compile> <port-to-curl> <expected-output-file>
+  options:
+    -d print out where tmp files are created and do not delete them
+    -h this help message
+
+  Note: all flags must occur before arguments
+"
+}
+
+DEBUG=false
```

Comment:
let's skip debug. It is always trivial to add set -x if you are debugging the script.